1.8.X fails to build on Fedora Rawhide for 32bit due to dcmtk

### Problem
During compilation on 32bit systems the following error occurs with the dicom library, dcmtk:
```
/builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp: In member function 'void OpenImageIO_v1_8::DICOMInput::read_metadata()':
/builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp:301:56: error: cannot bind non-const lvalue reference of type 'Sint32& {aka long int&}' to an rvalue of type 'Sint32 {aka long int}'
                 if (dataset->findAndGetSint32 (tag, val).good())
                                                        ^
In file included from /usr/include/dcmtk/dcmdata/dctk.h:50:0,
                 from /builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp:40:
/usr/include/dcmtk/dcmdata/dcitem.h:814:17: note:   initializing argument 2 of 'OFCondition DcmItem::findAndGetSint32(const DcmTagKey&, Sint32&, long unsigned int, bool)'
     OFCondition findAndGetSint32(const DcmTagKey &tagKey,
                 ^~~~~~~~~~~~~~~~
/builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp:305:56: error: cannot bind non-const lvalue reference of type 'Uint32& {aka long unsigned int&}' to an rvalue of type 'Uint32 {aka long unsigned int}'
                 if (dataset->findAndGetUint32 (tag, val).good())
                                                        ^
In file included from /usr/include/dcmtk/dcmdata/dctk.h:50:0,
                 from /builddir/build/BUILD/oiio-Release-1.8.7/src/dicom.imageio/dicominput.cpp:40:
/usr/include/dcmtk/dcmdata/dcitem.h:786:17: note:   initializing argument 2 of 'OFCondition DcmItem::findAndGetUint32(const DcmTagKey&, Uint32&, long unsigned int, bool)'
     OFCondition findAndGetUint32(const DcmTagKey &tagKey,
                 ^~~~~~~~~~~~~~~~
{standard input}: Assembler messages:
{standard input}: Error: .size expression for dicominput.cpp does not evaluate to a constant
```

Full build logs:
i686:
https://kojipkgs.fedoraproject.org//work/tasks/5776/23995776/build.log

armv7hl:
https://kojipkgs.fedoraproject.org//work/tasks/5778/23995778/build.log

### Versions

* OIIO branch/version:  1.8.7
* OS: Fedora Linux (Rawhide) 
* C++ compiler: 7.2.1

--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -478,7 +478,7 @@ endif()
 ###########################################################################
 # DCMTK
 if (USE_DICOM)
-    find_package (DCMTK)
+    find_package (DCMTK 3.6.1)
     if (NOT DCMTK_FOUND)
         set (DCMTK_INCLUDE_DIR "")
         set (DCMTK_LIBRARIES "")

--- a/src/cmake/modules/FindDCMTK.cmake
+++ b/src/cmake/modules/FindDCMTK.cmake
@@ -1,6 +1,3 @@
-# FIXME: CMake >= 3.5 has its own FindDCMTK.cmake, so when our minimum
-# cmake is at least that, we can remove this file.
-
 # Module to find DCMTK
 #
 # This module will first look into the directories defined by the variables:
@@ -8,9 +5,11 @@
 #
 # This module defines the following variables:
 #
-# DCMTK_FOUND       - True if DCMTK was found.
-# DCMTK_INCLUDES -    where to find DCMTK headers
-# DCMTK_LIBRARIES   - list of libraries to link against when using DCMTK
+# DCMTK_FOUND            True if DCMTK was found.
+# DCMTK_INCLUDES         Where to find DCMTK headers
+# DCMTK_LIBRARIES        List of libraries to link against when using DCMTK
+# DCMTK_VERSION          Version of DCMTK (e.g., 3.6.2)
+# DCMTK_VERSION_NUMBER   Int version of DCMTK (e.g., 362 for 3.6.2)
 
 include (FindPackageHandleStandardArgs)
 include (FindPackageMessage)
@@ -53,6 +52,13 @@ foreach (COMPONENT dcmimage dcmimgle dcmdata oflog ofstd iconv)
     endif ()
 endforeach()
 
+message (STATUS "DCMTK_INCLUDE_DIR = ${DCMTK_INCLUDE_DIR}")
+if (DCMTK_INCLUDE_DIR AND EXISTS "${DCMTK_INCLUDE_DIR}/dcmtk/config/osconfig.h")
+    file(STRINGS "${DCMTK_INCLUDE_DIR}/dcmtk/config/osconfig.h" TMP REGEX "^#define PACKAGE_VERSION[ \t].*$")
+    string(REGEX MATCHALL "[0-9.]+" DCMTK_VERSION ${TMP})
+    file(STRINGS "${DCMTK_INCLUDE_DIR}/dcmtk/config/osconfig.h" TMP REGEX "^#define PACKAGE_VERSION_NUMBER[ \t].*$")
+    string(REGEX MATCHALL "[0-9.]+" DCMTK_VERSION_NUMBER ${TMP})
+endif ()
 
 if (DCMTK_INCLUDE_DIR AND DCMTK_LIBRARIES)
     set(DCMTK_FOUND TRUE)
@@ -60,9 +66,22 @@ if (DCMTK_INCLUDE_DIR AND DCMTK_LIBRARIES)
     if (NOT DCMTK_FIND_QUIETLY)
         message(STATUS "Found DCMTK library ${DCMTK_LIBRARIES}")
         message(STATUS "Found DCMTK includes ${DCMTK_INCLUDES}")
+        message(STATUS "Found DCMTK short version number ${DCMTK_VERSION_NUMBER}")
     endif ()
 else()
     set(DCMTK_FOUND FALSE)
     message(STATUS "DCMTK not found. Specify DCMTK_PATH to locate it")
 endif()
 
+include (FindPackageHandleStandardArgs)
+find_package_handle_standard_args (DCMTK
+    REQUIRED_VARS   DCMTK_INCLUDE_DIR DCMTK_LIBRARIES
+    VERSION_VAR     DCMTK_VERSION
+    )
+
+mark_as_advanced (
+    DCMTK_INCLUDE_DIR
+    DCMTK_LIBRARIES
+    DCMTK_VERSION
+    DCMTK_VERSION_NUMBER
+    )

--- a/src/dicom.imageio/dicominput.cpp
+++ b/src/dicom.imageio/dicominput.cpp
@@ -297,13 +297,13 @@ DICOMInput::read_metadata ()
                     m_spec.attribute (name, (float)val);
                 // N.B. we cast to float. Will anybody care?
             } else if (evr == EVR_SL || evr == EVR_IS) {
-                int val;
+                Sint32 val;
                 if (dataset->findAndGetSint32 (tag, val).good())
-                    m_spec.attribute (name, val);
+                    m_spec.attribute (name, static_cast<int>(val));
             } else if (evr == EVR_UL) {
-                unsigned int val;
+                Uint32 val;
                 if (dataset->findAndGetUint32 (tag, val).good())
-                    m_spec.attribute (name, TypeDesc::UINT32, &val);
+                    m_spec.attribute (name, static_cast<unsigned int>(val));
             } else if (evr == EVR_US) {
                 unsigned short val;
                 if (dataset->findAndGetUint16 (tag, val).good())

--- a/src/dicom.imageio/dicominput.cpp
+++ b/src/dicom.imageio/dicominput.cpp
@@ -288,7 +288,7 @@ DICOMInput::read_metadata ()
                 if (dataset->findAndGetFloat32 (tag, val).good())
                     m_spec.attribute (name, val);
             } else if (evr == EVR_FD
-#if PACKAGE_VERSION_NUMBER >= 361
+#if PACKAGE_VERSION_NUMBER >= 362
                        || evr == EVR_OD
 #endif
                        ) {
@@ -312,7 +312,7 @@ DICOMInput::read_metadata ()
                        evr == EVR_DT || evr == EVR_LT || evr == EVR_PN ||
                        evr == EVR_ST || evr == EVR_TM || evr == EVR_UI ||
                        evr == EVR_UT || evr == EVR_LO || evr == EVR_SH
-#if PACKAGE_VERSION_NUMBER >= 361
+#if PACKAGE_VERSION_NUMBER >= 362
                        || evr == EVR_UC ||evr == EVR_UR
 #endif
                        ) {

